### PR TITLE
Enable simple local labels.

### PR DESCRIPTION
**Pull request by epi**
See added doc for details.

### CODE REVIEW

File: source/app.d [hunk 1]
```diff
@@ -261,21 +262,23 @@ void readSpaces() {
...
 		c &= 0xdf;
-		if (c >= 'A' && c <= 'Z') {
-			label ~= c;
+		if (c >= 'A' && c <= 'Z')
 			continue;
-		}
 		column--;
 		break;
 	}
+	string label = line[firstColumn .. column];
```
Comment (pfusik) [inline source/app.d]: The new lexing is likely faster. Unfortunately it makes the labels case-sensitive, which breaks compatibility.

**Review by pfusik — COMMENTED** [review]
I like the idea, but why question mark instead of, for example, a dot? Is it for compatibility with some other assembler?

**Comment by epi** [comment]
The choice of question mark was inspired by MAE, which I also followed in the old times of making hcasm. MADS has `opt +?` mode, in which it also claims to emulate MAE-style local labels.
ca65 uses `@`, but can be configured to a different character.
A dot would be intuitive for programmers of C-family languages, but might be confusing to users of other assemblers, as in many of them (including MAE, MADS and ca65) a dot starts a directive.

**Comment by pfusik** [comment]
Makes sense. Compatibility is good.

**Comment by epi** [comment]
Thank you!

**Comment by pfusik** [comment]
Tried in https://github.com/atari8xxl/unShrinkler/commit/1798a2b6455f40f2652faa2757d96c8e1fb95061. I like it!
Added `opt ?+` for MADS compatiblitiy: 52ef8c05d67a6f436cf94af51f38ac3bbc295497

**Pull request merged**